discli

```diff
@@ -121,11 +129,68 @@ class DiscourseHelper:
         os.remove(filename)
 
     def get_post(self, id):
-        post_path = urljoin(self.identity["url"], f"posts/{id}.json")
-        r = requests.get(post_path, params=self.api_auth)
-        if r.status_code == 200:
-            return json.loads(r.text)
-        raise RuntimeError("couldn't get post {}: status code {}", id, r.status_code)
+        response = self.request(f"posts/{id}.json")
+        if response.status_code == 200:
+            return json.loads(response.text)
+        raise RuntimeError("couldn't get post {}: status code {}", id, response.status_code)
+
+    def get_posts_in_category(self):
+        """
+        Get all the posts in a category
+        """
+
+        parser = argparse.ArgumentParser(description='get all posts in category')
+        parser.add_argument("category", help="name of the category to get posts from", type=str)
+        args = parser.parse_args(sys.argv[2:])
+
+        for page in range(1, 20):
+            print(f"Getting page {page} of results...")
+            response = self.request(f"/search.json?q=category%3A{args.category}&page={page}")
```

Comment:
It is possible to sort using `order:{something}` in your search results, and [the options for that are](https://github.com/discourse/discourse/blob/master/lib/search.rb#L576):

- "relevance" (default)
- latest post
- latest topic
- number of views
- number of likes

However, there doesn't seem to be any mechanism for reversing the order. So you can *only* get the latest first, not the oldest first.

A way to hack this might be to get the last page first and work your way backwards. Unfortunately, the [search API response](https://discourse.jujucharms.com/search.json?q=juju%20order:latest) itself doesn't tell you how many results there are, *but* the [categories API response](https://discourse.jujucharms.com/categories.json) does contain `topic_count` and `post_count` per category, so presumably one of these will match the number of search results for the query `category:docs`. So the way to do this would be:

- Check that the `topic_count` matches the total number of search results for `category:docs`
- Query the categories API to find this number
- Split it by the search results page size (50 I think) to infer the last page in the search results
- Search for the last page with `order:latest_topic`
- Work step backwards to page 1

I don't have time to implement this, but you're welcome to give it a go.